mm2src/mm2_bitcoin/spv_validation/src/conf.rs

```
@@ -0,0 +1,149 @@
+fn validate_btc_spv_header_height(coin: &str, height: u64) -> Result<(), SPVError> {
+    let is_retarget = height % RETARGETING_INTERVAL as u64;
+    if is_retarget != 0 {
+        return Err(SPVError::WrongRetargetHeight {
+            coin: coin.to_string(),
+            expected_height: height - is_retarget,
+        });
+    }
+
+    Ok(())
+}
+
+/// Custom SPV block header configuration
+#[derive(Clone, Debug, Deserialize)]
+pub struct BlockHeaderValidationParams {
+    pub difficulty_check: bool,
+    pub constant_difficulty: bool,
+    pub difficulty_algorithm: Option<DifficultyAlgorithm>,
+}
+
+#[derive(Debug, Clone, Deserialize)]
+pub struct SPVConf {
+    /// Where to start block headers sync from.
+    pub starting_block_header: SPVBlockHeader,
+    /// Max number of block headers to be stored in db, when reached, excessive headers will be deleted.
+    pub max_stored_block_headers: Option<NonZeroU64>,
+    /// The parameters that specify how the coin block headers should be validated. If None,
+    /// headers will be saved in DB without validation, can be none if the coin's RPC server is trusted.
+    pub validation_params: Option<BlockHeaderValidationParams>,
+}
+
+impl SPVConf {
+    pub fn validate(&self, coin: &str) -> Result<(), SPVError> {
+        if let Some(params) = &self.validation_params {
+            if let Some(algo) = &params.difficulty_algorithm {
+                match algo {
+                    DifficultyAlgorithm::BitcoinMainnet => {
+                        validate_btc_spv_header_height(coin, self.starting_block_header.height)?;
+                        if let Some(max) = self.max_stored_block_headers {
+                            validate_btc_max_stored_headers_value(max.into())?;
+                        }
+                    },
+                    DifficultyAlgorithm::BitcoinTestnet => {
+                        // Impl bitcoin testnet diffulty calculation.
+                        todo!()
```

Please return an error instead